common/djangoapps/student/views.py

```diff
@@ -1670,16 +1670,14 @@ def create_account_with_params(request, params):
         getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
     )
 
-    # Boolean of whether a 3rd party auth provider and credentials were provided in
-    # the API so the newly created account can link with the 3rd party account.
-    #
-    # Note: this is orthogonal to the 3rd party authentication pipeline that occurs
-    # when the account is created via the browser and redirect URLs.
-    should_link_with_social_auth = third_party_auth.is_enabled() and 'provider' in params
-
-    if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
+    if third_party_auth.is_enabled() and (pipeline.running(request) or 'provider' in params):
         params["password"] = pipeline.make_random_password()
 
+    # in case user is registering via third party and pipeline has expired, show appropriate
+    # error message
+    if 'is_third_party_request' in params and not pipeline.running(request):
+        raise ValidationError({'session_expired': ["Your session with third party has been expired"]})
```

Comment:
@srpearce Given @noraiz-anwar's diagnosis of the issue, the problem is that the registration process for the learner took longer than 10mns - and hence it timed out.  So, having them wait a few minutes is not necessary.  Instead, they need to re-register, but go through the process quicker.
For example, it may have taken them a while to find a unique username - so they kept trying over and over again to find one.  Or, the registration form asks too many questions that takes them a while to figure out.  Or, their network connection is too slow and the round trips to find a username took a while.  FYI @marcotuts